cmd/debugmailbox/src/lib.rs

```diff
@@ -64,21 +67,41 @@ enum DebugMailboxCmd {
     Debug,
     /// Force the device into ISP mode
     Isp,
+    /// Get a Debug Authentication Challenge
+    AuthChallenge { out: PathBuf },
+    /// Send a Debug Authentication Response
+    AuthResponse { dar: PathBuf },
 }
 
 fn alive<'a>(
     probe: &mut Box<dyn ArmProbeInterface + 'a>,
     addr: &ApAddress,
+    reset: bool,
 ) -> Result<()> {
-    probe.write_raw_ap_register(*addr, CSW, 0x21)?;
+    if reset {
+        probe.write_raw_ap_register(*addr, CSW, 0x21)?;
+    }
 
     let mut timeout = 100;
     loop {
         std::thread::sleep(std::time::Duration::from_millis(10));
-        if let Ok(val) = probe.read_raw_ap_register(*addr, CSW) {
-            if val == 0 {
-                break;
+        match probe.read_raw_ap_register(*addr, CSW) {
+            Ok(val) => {
+                println!("CSW = {val:x}");
+                if val == 0 {
+                    break;
+                }
+            }
+            Err(DebugProbeError::ArchitectureSpecific(arch_err)) => {
+                match arch_err.downcast::<DapError>() {
```

Comment:
This method for detecting and tolerating a WAIT appears in at least two places -- it'd be nice to collect it and give it a name.

Reply:
PTAL